### PR TITLE
hssi: support hssi performance

### PR DESCRIPTION
**Pull request by vmallavx**
Signed-off-by: =vmallavax <venkatax.visweswarachari.mallavarapu@intel.com>

### CODE REVIEW

File: samples/hssi/hssi_10g_cmd.h [hunk 1]
```diff
@@ -252,6 +267,96 @@ class hssi_10g_cmd : public hssi_cmd
...
+    uint32_t total_data_size_bits = (num_packets_  * data_pkt_size * 8) ;
+    double achieved_tx_tput_gbps = (total_data_size_bits / total_tx_duration_ns);
+    double achieved_rx_tput_gbps = (total_data_size_bits / total_rx_duration_ns);
+
+    std::cout << "\tSelected clock frequency : "<< clk_freq << " MHz" << std::endl
+    	      << "\tLatency minimum : " << latency_min_ns <<" ns"<<std::endl
+    	      << "\tLatency maximum : " << latency_max_ns <<" ns"<<std::endl
+    	      << "\tAchieved Tx throughput : " << achieved_tx_tput_gbps << " GB/s" << std::endl
+    	      << "\tAchieved Rx throughput : " << achieved_rx_tput_gbps << " GB/s" << std::endl
+    	      << std::endl;
```
Comment (tswhison) [inline samples/hssi/hssi_10g_cmd.h:357]: The remainder of the hssi application is written with the C++ API.  Why have we gone back to C here?

**Review by tswhison — COMMENTED** [review]
Thank you for your submission.  We need to keep the code consistent and change the C coded portion of your performance function using the OPAE C++ API please.  Also, can you comment on the availability of these performance registers in the current bitstreams?  Are these new registers always supported for the given bitstream IDs?

File: samples/hssi/hssi_10g_cmd.h [hunk 2]
```diff
@@ -252,6 +267,96 @@ class hssi_10g_cmd : public hssi_cmd
...
+    	return;
+    }
+
+    res = fpgaPropertiesSetObjectType(pprops, FPGA_DEVICE);
+    if (res != FPGA_OK) {
+    	std::cerr << "Failed to set type to FPGA_DEVICE" << std::endl;
+    	return;
+    }
+
+    fpgaEnumerate(&pprops, 1, &token, 1, &num_matches);
```
Comment (tswhison) [inline samples/hssi/hssi_10g_cmd.h:294]: Couple of issues here:

1. You're assuming that there is only 1 accelerator device in the system.  This is not a safe assumption.
2. You're ignoring the return value of fpgaEnumerate.  In general, that is not a safe thing to do.

File: samples/hssi/hssi_10g_cmd.h [hunk 3]
```diff
@@ -42,14 +42,27 @@
...
 
 #define CSR_MAC_LOOP          0x3e00
 
+#define ETHERNET_SPEED        10  // Gbps
+
+#define USER_CLKFREQ_S10      156.25  // MHz
+#define USER_CLKFREQ_AC       300.00  // MHz
+#define BITSVER_MAJOR_S10     4
+#define BITSVER_MAJOR_AC      5
+#define DFL_BITSTREAM_ID      "bitstream_id"
```
Comment (tswhison) [inline samples/hssi/hssi_10g_cmd.h:62]: This is unused and can be removed.

**Comment by tswhison** [comment]
If you like, I can help you with the enumeration changes.  Please let me know if we can work together on your branch.

**Review by tswhison — APPROVED** [review]

File: samples/hssi/hssi_cmd.h [hunk 4]
```diff
@@ -33,10 +33,20 @@
...
+namespace fpga = opae::fpga::types;
 
-#define PKT_FILT_CSR_DEST_ADDR 0x0040
+#define INVALID_MAC           0xffffffffffffffffULL
+
+#define INVALID_CLOCK_FREQ    static_cast<double>(0.0)
 
-#define INVALID_MAC 0xffffffffffffffffULL
+#define USER_CLKFREQ_S10      static_cast<double>(156.25)  // MHz
+#define USER_CLKFREQ_N6000    static_cast<double>(350.00)  // MHz
```
Comment (r-rojo) [inline samples/hssi/hssi_cmd.h:43]: why the static casts in the  #defiines? can't we use literals?

**Review by r-rojo — DISMISSED** [review]

**Review by r-rojo — APPROVED** [review]

**Comment by tswhison** [comment]
Hi [vmallavx](https://github.com/vmallavx), please close this if it is no longer valid.  Thanks

**Pull request closed without merging**